### PR TITLE
feat: add digest to artifacts info of published docker images

### PR DESCRIPTION
**Pull request by gal-legit**
Extract the digest (sha256) of docker images from the `docker push` command for dockers published to the docker registry.
Outputting the digest is required to avoid a race condition when referencing the image, where the image tag is being modified before the reference is done.
See this [blog post](https://github.com/goreleaser/goreleaser/issues/3496) for more info.
This PR fixes https://github.com/goreleaser/goreleaser/issues/3496.

Note that the 'publish' pipe now must run before the 'metadata' pipe, so that the information extracted during the 'publish' pipe would appear in the metadata.
Previously, the published docker images metadata wasn't printed (because of the order). It made sense because the content of the published image was just a subset of the local one.
Now that it is printed to the metadata, it should have a different name to avoid confusion.
As I mentioned, it wasn't printed before - so there shouldn't be any backward-compatibility issues.

---
Local tests:
```
go test -v .
=== RUN   TestVersion
=== RUN   TestVersion/only_version
=== RUN   TestVersion/version_and_date
=== RUN   TestVersion/version,_date,_built_by
=== RUN   TestVersion/all_empty
=== RUN   TestVersion/complete
--- PASS: TestVersion (0.00s)
    --- PASS: TestVersion/only_version (0.00s)
    --- PASS: TestVersion/version_and_date (0.00s)
    --- PASS: TestVersion/version,_date,_built_by (0.00s)
    --- PASS: TestVersion/all_empty (0.00s)
    --- PASS: TestVersion/complete (0.00s)
PASS
ok      github.com/goreleaser/goreleaser        0.764s
```

Output example:
```
  {
    "name": "gallegit/hello-world:latest",
    "path": "gallegit/hello-world:latest",
    "goos": "linux",
    "goarch": "amd64",
    "internal_type": 10,
    "type": "Published Docker Image",
    "extra": {
      "digest": "sha256:c3f7dd196a046dc061236d3c6ae1e2946269e90da30b0a959240ca799750e632"
    }
  }
```

### CODE REVIEW

**Comment by caarlos0** [comment]
hey @gal-legit ! this looks awesome!

can you cherry-pick this commit? https://github.com/goreleaser/goreleaser/commit/d389030f5c89949a2d3b102eb9d9c7610a6fc0f2

so we ensure this is not broken at some point :)

File: internal/pipe/docker/api.go [hunk 1]
```diff
@@ -66,3 +66,31 @@ func runCommand(ctx *context.Context, dir, binary string, args ...string) error
...
+
+	var b bytes.Buffer
+	w := gio.Safe(&b)
+	cmd.Stderr = io.MultiWriter(logext.NewWriter(fields, logext.Error), w)
+
+	log.WithFields(fields).WithField("args", args[1:]).Debug("running")
+	out, err := cmd.Output()
+	if out != nil {
+		// regardless of command success, always print stdout for backward-compatibility with runCommand()
+		io.MultiWriter(logext.NewWriter(fields, logext.Error), w).Write(out)
```
Comment (caarlos0) [inline internal/pipe/docker/api.go:89]:
```suggestion
		_, _ = io.MultiWriter(logext.NewWriter(fields, logext.Error), w).Write(out)
```

**Comment by gal-legit** [comment]
@caarlos0 done. thank for the feedback! :)

**Comment by caarlos0** [comment]
awesome, thank you @gal-legit !

**Pull request merged**